packages/compartment-mapper/test/test-bundle.js

```diff
@@ -77,3 +77,49 @@ test('equivalent archive behaves the same as bundle', async t => {
   });
   t.deepEqual(log, expectedLog);
 });
+
+// This is failing because it requires support for missing dependencies.
+// Cannot bundle: encountered deferredError Cannot find file for internal module "./spam"
+test.failing('bundle cjs-compat', async t => {
+  const cjsFixture = new URL(
+    'fixtures-cjs-compat/node_modules/app/index.js',
+    import.meta.url,
+  ).toString();
+
+  const bundle = await makeBundle(read, cjsFixture);
+  const log = [];
+  const print = entry => {
+    log.push(entry);
+  };
+  const compartment = new Compartment({ print });
+  compartment.evaluate(bundle);
+  t.deepEqual(log, expectedLog);
+});
+
+test('bundle cjs-compat default-difficulties', async t => {
+  const cjsFixture = new URL(
+    'fixtures-cjs-compat/node_modules/default-difficulties/index.mjs',
+    import.meta.url,
+  ).toString();
+
+  const bundle = await makeBundle(read, cjsFixture);
+  fs.writeFileSync('bundle.js', bundle);
```

Comment:
As a matter of tending toward web-portable ocap style, you could `await writePowers.write` here.

Reply:
oh this was not supposed to make it in, thanks for catching it